framework/DataObjects/XPointSet.py

```diff
@@ -168,3 +168,41 @@ def _selectiveRealization(self,rlz):
     for var in toRemove:
       del rlz[var]
     return rlz
+
+  def _toCSV(self,fileName,start=0,**kwargs):
+    """
+      Writes this data object to CSV file (except the general metadata, see _toCSVXML)
+      @ In, fileName, str, path/name to write file
+      @ In, start, int, optional, first realization to start printing from (if > 0, implies append mode)
+      @ In, kwargs, dict, optional, keywords for options
+            Possibly includes:
+                'clusterLabel': name of variable to cluster printing by.  If included then triggers history-like printing.
+      @ Out, None
+    """
+    if not self.hierarchical and 'RAVEN_isEnding' in self.getVars():
```

Comment:
Again, comments here explaining that we only need to specialize if not "self.hierarchical" but RAVEN_isEnding would help understand why there's a a specialization here.  Also the purpose of "self.hierarchical" may not be self-explanatory.

Reply:
added